docs/haskell-use-cases.rst

```diff
@@ -884,3 +884,127 @@ not possible to build docker images for haskell binaries as above using rules_do
 
 Following these steps you should end up with a fairly lightweight docker image, bringing the flexibility of nix
 as a docker base image manager and the power of ``rules_haskell`` for your haskell build together.
+
+Cross-compilation
+-----------------
+
+Currently, ``rules_haskell`` only supports cross-compiling to ``arm`` on linux.
+Cross-compiling requires providing a cross-compiler, telling ``rules_haskell``
+about it, and then requesting Bazel to build for the target platform.
+
+Ideally, providing a cross-compiler would only require the advice in
+`Picking a compiler`_. However, the case of ``arm`` requires to configure
+a few aspects at this time. One has to make available the llvm tools
+to the compiler, emulation support needs to be set to enable
+compilation of Template Haskell splices via an external interpreter,
```

Comment:
Can this be made optional, say if the user does not use TH?

Reply:
It isn't needed if TH is not used, indeed.

Right now, the user can remove the `-fexternal-interpreter` flag in `arm-cross.nix` if she doesn't want TH. GHC doesn't start the interpreter unless some module uses TH, so it doesn't harm to leave the nix configuration as is if some targets use TH and others don't.